### ISSUE TITLE
Add RateLimiter.createBurstingLimiter() with variable burst bucket

### ISSUE THREAD

**Issue by amcp** [google/guava#3112]
Currently, the constructor with a variable burst bucket size, [`SmoothRateLimiter.SmoothBursty(SleepingStopwatch stopwatch, double maxBurstSeconds)`](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/SmoothRateLimiter.java#L280) is package private. The public static factory methods [hard code a burst bucket size of 1](https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/RateLimiter.java#L132). There are use cases where a burst bucket [larger than 1 second are needed](https://github.com/awslabs/dynamodb-janusgraph-storage-backend/blob/master/src/main/java/com/google/common/util/concurrent/RateLimiterCreator.java#L27). Therein linked is a candidate implementation and [a corresponding test case](https://github.com/awslabs/dynamodb-janusgraph-storage-backend/blob/master/src/test/java/com/google/common/util/concurrent/RateLimiterCreatorTest.java#L31). I would like to merge this addition upstream.

**Comment by nikhilbarar** [comment]
Can i pick this?

**Comment by amcp** [comment]
Yes please do whatever necessary to get included in guava upstream. Mention/Add me to the PR so I can review.
